src/fsharp/FSharp.Core/map.fs

```diff
@@ -538,6 +538,26 @@ module MapTree =
           interface System.IDisposable with 
               member _.Dispose() = ()}
 
+    let rec leftmost m =
+        if isEmpty m then 
+            throwKeyNotFound()
```

Comment:
I wonder whether `KeyNotFoundException` is the proper exception to rise here. We only use it in `find`. Although it's _technically_ correct, it may be confusing for the user. Shall we have a separate exception for traversal-like cases, when map is empty? thoughts @dsyme?